Report/Template.Rmd

```diff
@@ -243,17 +251,262 @@ datasets = read.and.clean(dataset = "easySHARE_rel6_0_0.rda", wav = 99)
 ```
 
 \section{Multidisciplinary and crossnational summary statistics}
-JULIAN
-\subsection{Theory and Design}
-\subsection{Implementation}
-\subsection{Empirical Results}
+
+## Theory and Design
+
+Since we are working with a cross-national survey, it is useful to to present an overview on a country-by-country basis. For each country, survey participants will be classified according to their age group. Within each class, shares of gender-specific as well as overall labor participation rates and health indicators are calculated and discussed. 
+
+
+## Implementation
+
+An efficient way of seperating the easySHARE dataset into country, gender and age is explained in the following. In order to minimize hard-coding and possibly related errors, a grid is expanded between named variables.
+Functions that calculate group shares per country and labor participation rates can subsequently be applied onto the defined grid. 
+
+Besides the target data frame, `group.share` takes a boolean variable as an input that specifies whether aggregation on a national level is intended. The share of males or females (per country) is calculated by finding the sum of occurences and dividing by their total, represented by the length of the vector. A similar procedure that allows for multiple subsets and based upon `dplyr` is used in `labor`.
+
+```{r, eval = FALSE}
+# Function for calculation group shares per country
+group.share = function(df, y, agg.country = TRUE) {
+    if (agg.country) {
+        final = tapply(df[[y]], df$country, function(x) {
+            val = sum(x)/length(x)
+            return(val)
+        })
+    } else { # return total if country-level aggregation not specified
+        final = sum(df[[y]])/length(df[[y]])
+    }
+    return(final)
+}
+
+# Function for calculating labor participation rate per country / gender / age
+labor = function(df, .country, .gender, .age) {
+    dataset = df %>% 
+        filter(country == .country & gender == .gender & get(.age) == TRUE)
+    denom = nrow(dataset)
+    numer = with(dataset, sum(labor_ft | labor_pt))
+    perc  = round(numer/denom, 4)
+    
+    return(perc)
+}
+
+```
+
+Within the code below, `args.list` contains three sub-lists divided by countriy (`c`), gender (`g`) and age (`a`). The list is passed as an argument to `labor`, which are simulatenously iterated over the sub-lists using `pmap`. This function has an advantage over a similar functions, since it allows multiple arguments and uses parallelization by default.  
+Including `with` to specify the data frame on which we expand the grid upon, modifies the data frame in a new environment. This simplifies name assignment. 
```

Comment:
I mentioned this the other day, but `with` just allows you to apply an expression (in our case `expand.grid`) over a data frame. So the advantage is really just fewer lines of code. Maybe we don't need to put this.

Reply:
yep, deleted :)